src/libraries/System.IO.Ports/src/System/IO/Ports/SerialStream.cs

```diff
@@ -10,9 +10,6 @@
 
 namespace System.IO.Ports
 {
-// Issue https://github.com/dotnet/runtime/issues/54916
```

Comment:
What about the Windows part? I know it's very archaic, but it seems that the implementation uses `ReadFile` and `WriteFile` methods, which means that we could just create a `FileStream`  (with buffering disabled by passing `bufferSize: 1`) from the `SafeFileHandle` we get in https://github.com/dotnet/runtime/blob/6fc8441311120c47c96b2605bf3a0d186f82454a/src/libraries/System.IO.Ports/src/System/IO/Ports/SerialStream.Win32.cs#L11-L13 and delegate all reads and writes to `FileStream`.

Reply:
I'll take a look, but it feels like a separate PR to me..